src/smhi/metobs.py

```diff
@@ -323,6 +323,9 @@ def __init__(
         """
         super().__init__()
 
+        if period == "not-set":
+            period = periods_in_station.data[0]
```

Comment:
But doesn't this mean it is always sorted? And chooses in the other order? Anyway, you are right, I will look in to it and see if I can fix this.

Reply:
I couldn't tell that there was ever any case in which the data would be sorted in another manner than corrected-archive, latest-month,latest-day,latest-hour, so I added an additional sorting step just in case. Won't really affect performance anyhow :)